code42_connector.py

```diff
@@ -38,29 +38,19 @@ def __init__(self):
         self._password = None
         self._client = None
         self._action_map = {
-            self.TEST_CONNECTIVITY_ACTION_ID: lambda x: self._handle_test_connectivity(x),
-            self.ADD_DEPARTING_EMPLOYEE_ACTION_ID: lambda x: self._handle_add_departing_employee(x),
-            self.REMOVE_DEPARTING_EMPLOYEE_ACTION_ID: lambda x: self._handle_remove_departing_employee(x),
-            self.LIST_DEPARTING_EMPLOYEES_ACTION_ID: lambda x: self._handle_list_departing_employees(x),
-            self.ADD_HIGH_RISK_EMPLOYEE_ACTION_ID: lambda x: self._handle_add_high_risk_employee(x),
-            self.REMOVE_HIGH_RISK_EMPLOYEE_ACTION_ID: lambda x: self._handle_remove_high_risk_employee(x),
-            self.LIST_HIGH_RISK_EMPLOYEES_ACTION_ID: lambda x: self._handle_list_high_risk_employees(x)
+            self.TEST_CONNECTIVITY_ACTION_ID: self._handle_test_connectivity,
```

Comment:
What about just having the action map be a decorator on each function instead of enumerating them all?:

```
ACTION_MAP = {}

def action_key(key):
    def wrapper(f):
        ACTION_MAP[key] = f
        return f

    return wrapper
```
Less stuff to write and it ties the key closely to the method itself: 

```
    @action_key("test_connectivity")
    def _handle_test_connectivity(self, param):
```

Reply:
I _realllllllly_ like this idea, it makes the association between the key and the function it's tied to super clear

Reply:
I like this way this looks!